nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/connect/LauncherRegisterHelper.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nuxeo
+ */
+
+package org.nuxeo.launcher.connect;
+
+import java.io.Console;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.nuxeo.connect.NuxeoConnectClient;
+import org.nuxeo.connect.registration.RegistrationException;
+
+/**
+ * @author <a href="mailto:ak@nuxeo.com">Arnaud Kervern</a>
+ * @since 8.2
+ */
+public class LauncherRegisterHelper {
+    public static String[] promptPasswords() throws IOException {
+        String[] ret = new String[2];
+
+        Console console = System.console();
+
+        if (console != null) {
+            ret[0] = new String(console.readPassword("Please enter your password: "));
+            ret[1] = new String(console.readPassword("Please re-enter your password: "));
+        } else {
+            String password = new String(IOUtils.toByteArray(System.in)).trim();
+            ret = new String[] { password, password };
+        }
+        return ret;
+    }
+
+    public static boolean promptAcceptTerms() {
+        Console console = System.console();
+        if (console != null) {
+            String terms;
+            do {
+                terms = console.readLine("Read and accept the Nuxeo Trial Terms and Conditions - https://www.nuxeo.com/en/services/nuxeo-trial-tc (y/n): ");
+            } while (!(terms.startsWith("y") || terms.startsWith("n")));
+
+            return terms.startsWith("y");
+        } else {
+            return true;
```

Comment:
If not interactive, then the user is supposed to have accepted the terms? If so, it's worth to display an informational message.